plugins/providers/docker/provider.rb

```diff
@@ -11,6 +11,14 @@ module DockerProvider
     class Provider < Vagrant.plugin("2", :provider)
       @@host_vm_mutex = Mutex.new
 
+      def self.usable?(raise_error=false)
+        Driver.new.execute("docker", "version")
+        true
+      rescue Vagrant::Errors::CommandUnavailable, Errors::ExecuteError
```

Comment:
I _think_ we will also need to rescue on `Vagrant::Errors::CommandUnavailableWindows` here

Reply:
Not necessary, because `CommandUnavailableWindows` [inherits](https://github.com/hashicorp/vagrant/blob/v2.2.4/lib/vagrant/errors.rb#L315) from `CommandUnavailable`. I have tested this on Linux and Windows.